lib/Db/ProvisioningMapper.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2021 Anna Larch <anna@nextcloud.com>
+ *
+ * @author 2021 Anna Larch <anna@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Db;
+
+use OCP\AppFramework\Db\QBMapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+
+/**
+ * @template-extends QBMapper<Provisioning>
+ */
+class ProvisioningMapper extends QBMapper {
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, 'mail_provisionings');
+	}
+
+	public function getActive() : ?array {
+		$qb = $this->db->getQueryBuilder();
+		$qb = $qb->select('*')
+				->from($this->getTableName())
+				->where($qb->expr()->eq('active', $qb->createNamedParameter('true', IQueryBuilder::PARAM_BOOL)));
+		$active = $qb->execute();
+		$result = $active->fetchAll();
+		$active->closeCursor();
+		return $result;
+	}
+
+	public function getAll() : ?array {
+		$qb = $this->db->getQueryBuilder();
+		$qb = $qb->select('*')
+				->from($this->getTableName());
+		$all = $qb->execute();
+		$result = $all->fetchAll();
+		$all->closeCursor();
+		return $result;
+	}
+
+	public function getWildcardConfig() : ?array {
+		$qb = $this->db->getQueryBuilder();
+		$qb = $qb->select('*')
+				->from($this->getTableName())
+				->where($qb->expr()->eq('provisioning_domain', $qb->createNamedParameter('*', IQueryBuilder::PARAM_BOOL)));
+		$wildcard = $qb->execute();
+		$result = $wildcard->fetch();
```

Comment:
``\OCP\AppFramework\Db\QBMapper::findEntities``

Reply:
It's in the Refactoring To Do ;)